### PR TITLE
Adds description of unit testing Pyspark code + samples

### PR DESCRIPTION
**Pull request by nairmai**
* Adds documentation of unit testing Pyspark code using `pytest` (+ usage of fixtures, data factories, etc.)
* Adds sample test suite and `conftest.py`
* Removes extraneous test files

Co-authored with @jakereifer, @evgri243

### CODE REVIEW

File: src/python/src/tests/batch_operations/test_filter_by_supplier.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    df = message_factory(supplierId="Sup1")
+
+    res = filter_by_supplier(df, "Sup2")
+
+    assert res.count() == 0
+
+@pytest.mark.parametrize("supplierId, expected",
+    [
+        pytest.param('-1', '-1', id="Keeps supplier -1"),
+        pytest.param('0', '0', id="Keeps supplier -1")
```
Comment (jakereifer) [inline src/python/src/tests/batch_operations/test_filter_by_supplier.py:44]: The id here is probably a copy/paste issue?

Comment (nairmai) [reply src/python/src/tests/batch_operations/test_filter_by_supplier.py]: Fixed, thanks!

**Comment by evgri243** [comment]
@MarekLani, I was required to move unit-tests to src folder to make local execution easier. I've adjusted entrypoint.sh, is it all what is needed?

**Comment by MarekLani** [comment]
> @MarekLani, I was required to move unit-tests to src folder to make local execution easier. I've adjusted entrypoint.sh, is it all what is needed?

yes it is all

**Review by evgri243 — APPROVED** [review]

**Pull request merged**